### ISSUE TITLE
gadgets: Run local tests on cri-o too

### ISSUE THREAD

**Issue by mauriciovasquezbernal** [inspektor-gadget/inspektor-gadget#3175]
Currently we run the local tests for the gadgets on docker and containerd.  We should also support cri-o for this. 
#3172 added support for containerd, however the cri-o will need more work as the container factory (pkg/testing/containers/factory.go) doesn't support cri-o.

**Comment by Snehil-Shah** [comment]
I am working on this. Can I be assigned?

**Comment by mauriciovasquezbernal** [comment]
> I have a doubt, I would also have to write test utils to initialize and run a new cri-o container right? Can I get some pointers on how can I write them?

Exactly, the idea is to have a way to start cri-o containers from the test (see how it's done for docker, containerd, etc.).

For cri-o, I think you can get some inspiration from [crictl](https://github.com/kubernetes-sigs/cri-tools/tree/master/cmd/crictl), like here https://github.com/kubernetes-sigs/cri-tools/blob/master/cmd/crictl/container.go.

**Comment by farhanashrafdev** [comment]
anyone still working on it? can I work on it?
